src/dbs/mod.rs

```diff
@@ -133,4 +305,141 @@ mod tests {
 				.hash
 		)
 	}
+
+	#[test(tokio::test)]
+	async fn test_capabilities() {
+		// (Capabilities, Query, Succeeds, Response Contains)
+		let cases = vec![
+			//
+			// Functions and Networking is allowed
+			//
+			(
+				Capabilities::default(),
+				"RETURN http::get('http://127.0.0.1')",
+				false,
+				"Connection refused",
+			),
+			//
+			// Scripting is allowed
+			//
+			(Capabilities::default(), "RETURN function() { return '1' }", true, "1"),
+			//
+			// Scripting is not allowed
+			//
+			(
+				Capabilities::default().with_scripting(false),
+				"RETURN function() { return '1' }",
+				false,
+				"Scripting functions are not allowed",
+			),
+			//
+			// Some functions are not allowed
+			//
+			(
+				Capabilities::default()
+					.with_allow_funcs(Targets::<FuncTarget>::Some(
+						[FuncTarget::from_str("http::*").unwrap()].into(),
+					))
+					.with_deny_funcs(Targets::<FuncTarget>::Some(
+						[FuncTarget::from_str("http::get").unwrap()].into(),
+					)),
+				"RETURN http::get('http://127.0.0.1')",
+				false,
+				"Function 'http::get' is not allowed",
+			),
+			(
+				Capabilities::default()
+					.with_allow_funcs(Targets::<FuncTarget>::Some(
+						[FuncTarget::from_str("http::*").unwrap()].into(),
+					))
+					.with_deny_funcs(Targets::<FuncTarget>::Some(
+						[FuncTarget::from_str("http::get").unwrap()].into(),
+					)),
+				"RETURN http::head('http://127.0.0.1')",
+				false,
+				"Connection refused",
```

Comment:
This assumes user running the test suite doesn't have a web server running locally on port 80.